clang/test/SemaCXX/warn-unsafe-buffer-usage-array.cpp

```diff
@@ -124,3 +124,36 @@ void array_indexed_const_expr(unsigned idx) {
   k = arr[get_const(5)]; // expected-note {{used in buffer access here}}
   k = arr[get_const(4)];
 }
+
```

Comment:
nit: maybe add a comment like `// Test the case where indexes of the const array access evaluate to constants and depend on template arguments.`?